flask/app/participants.py

```diff
@@ -37,38 +37,65 @@
 def list_participants():
 
     # parsing query parameters
-    max_rows = request.args.get("max_rows", default=100)
-    starts_with = request.args.get("starts_with", default="", type=str)
-    starts_with = f"{starts_with}%"  # sql syntax
-    order_by_col = request.args.get("order", default="participant_id", type=str)
-    # need some default or we need an ifelse statement, one with order_by method and one without. AFAIK there is no 'default' parameter we can pass into order_by to get the default sql ordering scheme
+    limit = request.args.get("limit", default=10)
+    page = request.args.get("page", default=1)
+    order_by_col = request.args.get("order_by", default="participant_id", type=str)
+    order_dir = request.args.get("order_dir", default="asc", type=str)
 
+    # for some reason type=int doesn't catch non-integer queries
     try:
-        int(max_rows)
+        int(limit)
     except:
-        return "Max rows must be a valid integer", 400
+        return "Limit must be a valid integer", 400
+
+    try:
+        int(page)
+    except:
+        return "Page must be a valid integer", 400
+
+    offset = (int(page) * int(limit)) - int(limit)
 
     columns = models.Participant.__table__.columns.keys()
 
-    if order_by_col not in columns:
-        return f"Column name for ordering must be one of {columns}", 400
-    column = getattr(models.Participant, order_by_col)
+    raw_filters = request.args.getlist("filter")
+    if raw_filters:
+        filt = filter_query(models.Participant, raw_filters)
```

Comment:
Is it possible to pass in the joined tables as the model here? We should be able to filter by any column that gets sent out at the end (eg. family_codename, dataset_types)